metamorph/processor.go

```diff
@@ -225,62 +216,48 @@ func (p *Processor) processMinedCallbacks() {
 	}()
 }
 
-func (p *Processor) processStatusUpdates() {
+func (p *Processor) CheckAndUpdate(statusUpdatesMap *map[chainhash.Hash]store.UpdateStatus, skipCheck bool) {
+	if (skipCheck && len(*statusUpdatesMap) < p.processStatusUpdatesBatchSize) || len(*statusUpdatesMap) == 0 {
```

Comment:
I find it a good idea to extract the duplicate code into this function. But I think this part I would keep out of it. Because that is a bit hard to read.
```
if (skipCheck && len(*statusUpdatesMap) < p.processStatusUpdatesBatchSize) || len(*statusUpdatesMap) == 0 {}
```

Then you don't need the `skipCheck` parameter. 

With this I mean I would keep this code before calling `CheckAndUpdate`
```
if len(statusUpdatesMap) < p.processStatusUpdatesBatchSize {
					continue
				}
```
... and this one respectively
```
				if len(statusUpdatesMap) == 0 {
					continue
				}
```